### ISSUE TITLE
Kaggle Dataset: validation and testing methodology

### ISSUE THREAD

**Issue by rajesh-iiith** [facebookresearch/dlrm#192]
The DLRM paper says that _"The Criteo Ad Kaggle data set contains approximately 45 million samples over 7 days. In experiments, typically the 7th day is split into validation and test set while the first 6 days are used as the training set."_

I have two questions:

1. I don't see the output labels on the test data (7th-day data). Where can I get that?
2. The clicked-unclicked distribution in the test set is around 25%-75%. So, even an all-unclicked classifier would give an accuracy of 75%. Paper reports the validation accuracy of 79%, which is not substantially higher than 75%. How are you taking care of this skewness in your validation methodology?

**Comment by huwan** [comment]
Let me just add a few comments on your first question:
- train.txt in the criteo ad kaggle data set itself contains samples over a period of 7 days, and it will be split into training and test parts.
- test.txt in the criteo ad kaggle data set corresponds to samples on the day following the training period. But it is not used in the experiments because its label remains officially undisclosed.

PS: I am NOT one of the authors.

**Comment by rajesh-iiith** [comment]
> Let me just add a few comments on your first question:
> 
> * train.txt in the criteo ad kaggle data set itself contains samples over a period of 7 days, and it will be split into training and test parts.
> * test.txt in the criteo ad kaggle data set corresponds to samples on the day following the training period. But it is not used in the experiments because its label remains officially undisclosed.
> 
> PS: I am NOT one of the authors.

@huwan , How to know from where the data of 7th day begins?

**Comment by huwan** [comment]
There is no timestamp given in the dataset, so it's impossible to know the exact position from where the data of 7th day begins.

DLRM splits inputs from train.txt into 7 sub-datasets of approximately equal size. We can get the offset of each day easily. Please refer to code [here](https://github.com/facebookresearch/dlrm/blob/master/dlrm_data_pytorch.py#L123-L131) in dlrm_data_pytorch.py (pytorch version).

**Comment by mnaumovfb** [comment]
Thank you for the summary on the days in the dataset @huwan. 

The increase from 75% to 79% can be considered significant for this dataset. Typically you can use different metrics while training on skewed datasets. For instance, you can use AUC rather than accuracy to see the progress of the model training.

**Comment by mnaumovfb** [comment]
The metric reported in DLRM paper is BCE loss and accuracy. 

However, in DLRM you can choose to track different metrics, including AUC, see https://github.com/facebookresearch/dlrm/blob/master/dlrm_s_pytorch.py#L821

**Comment by mnaumovfb** [comment]
It seems that this issue is resolved. Closing.
